d3a_api_client/aggregator.py

```diff
@@ -33,8 +34,10 @@ def _handle_event_message(self, message):
 
 class Aggregator(RestDeviceClient):
 
-    def __init__(self, aggregator_name, simulation_id=None, domain_name=DOMAIN_NAME_FROM_ENV,
-                 websockets_domain_name=WEBSOCKET_DOMAIN_NAME_FROM_ENV, accept_all_devices=True):
+    def __init__(self, aggregator_name, simulation_id=simulation_id_from_env(),
+                 domain_name=domain_name_from_env(),
+                 websockets_domain_name=websocket_domain_name_from_env(),
```

Comment:
I think it is a good idea to change the current behavior. Some special attention is needed in this case, as you would be using callables as default arguments.
> Python’s default arguments are evaluated once when the function is defined, not each time the function is called

This means that the default value returned by these functions would always be the same, even if the user decided to change the environment variable at some point. See this example:
```python
import os
os.environ['SOME_SETTING_VALUE'] = 'my cool content'

def callable_function():
    return os.getenv('SOME_SETTING_VALUE')

callable_function()
# 'my cool content'

def function_with_callable_defaults(content=callable_function()):
    print(content)

function_with_callable_defaults()
# my cool content

# Now we change the environment variable
os.environ['SOME_SETTING_VALUE'] = 'some new different content'

function_with_callable_defaults()
# my cool content
```
As you see, the content of the default argument didn't change! This can get quite nasty unfortunately.

Reply:
One common way to overcome this would be to do something like this:
```suggestion
    def __init__(self, aggregator_name, simulation_id=None, ...):
             simulation_id = simulation_id if simulation_id else simulation_id_from_env()
```

Reply:
Actually, the result of the callable (`simulation_id_from_env()`) IS evaluated each time the `__init__` method is called. It was the whole motivation behind my change.
However, will change it to your suggestion as it more explicite.

Reply:
Wait, are you sure about this? This is surprising to me and I can't reproduce it:
```python
import os
os.environ['SOME_SETTING_VALUE'] = 'my cool content'

def callable_function():
    return os.getenv('SOME_SETTING_VALUE')

callable_function()
# 'my cool content'

class AggregatorFake:
    def __init__(self, simulation_id=callable_function()):
        self.simulation_id = simulation_id

AggregatorFake().simulation_id
# 'my cool content'

# Now we change the environment variable
os.environ['SOME_SETTING_VALUE'] = 'some new different content'

AggregatorFake().simulation_id
# 'my cool content'  (not updated)
```

Reply:
weird, you are right, can not reproduce it, too. Will go your proposed way.